src/cache.js

```diff
@@ -0,0 +1,131 @@
+import window from 'global/window';
+import {getUrl} from './url';
+
+// Tracks previously-downloaded scripts and/or detected players.
+//
+// The keys follow the format "accountId_playerId_embedId" where accountId is
+// optional and defaults to "*". This happens when we detect pre-existing
+// player globals.
+const cache = new window.Map();
+
+/**
+ * Get the cache key given some properties.
+ *
+ * @private
+ * @param  {object} props
+ *         Properties describing the player record to cache.
+ *
+ * @param  {string} props.playerId
+ *         A player ID.
+ *
+ * @param  {string} props.embedId
+ *         An embed ID.
+ *
+ * @param  {string} [props.accountId="*"]
+ *         An optional account ID. This is optional because when we search for
+ *         pre-existing players to avoid downloads, we will not necessarily
+ *         know the account ID.
+ *
+ * @return {string}
+ *         A key to be used in the script cache.
+ */
+const key = ({accountId, playerId, embedId}) => `${accountId || '*'}_${playerId}_${embedId}`;
+
+/**
+ * Add an entry to the script cache.
+ *
+ * @private
+ * @param  {object} props
+ *         Properties describing the player record to cache.
+ *
+ * @param  {string} props.playerId
+ *         A player ID.
+ *
+ * @param  {string} props.embedId
+ *         An embed ID.
+ *
+ * @param  {string} [props.accountId="*"]
+ *         An optional account ID. This is optional because when we search for
+ *         pre-existing players to avoid downloads, we will not necessarily
+ *         know the account ID. If not given, we assume that no script was
+ *         downloaded for this player.
+ *
+ * @return {string}
+ *         A key to be used in the script cache.
+ */
+const store = (props) => {
```

Comment:
Should `store` be changed to `set`? Seems more consistent with the API names for a Map. Then make it take two arguments, and default it to the current default.

Reply:
> default it to the current default
I don't follow your meaning?

Reply:
Also, I'm not sure what being consistent with `Map` gives us. This is supposed to be a cache of script URLs and identifiers, doesn't matter if it's a `Map` internally.